vic/drivers/cesm/cpl_mct/lnd_comp_mct.F90

```diff
@@ -0,0 +1,729 @@
+                                       shr_file_setLogUnit, shr_file_setIO
+    USE seq_timemgr_mod       , ONLY : seq_timemgr_EClockGetData
+    USE mct_mod               , ONLY : mct_gsMap, mct_gGrid, mct_gGrid_importIAttr, &
+                                       mct_gGrid_importRAttr, mct_gGrid_init, &
+                                       mct_gsMap_orderedPoints
+    USE seq_infodata_mod      , ONLY : seq_infodata_PutData, seq_infodata_PutData
+    USE seq_flds_mod
+    USE vic_cesm_def_mod
+    USE vic_cesm_print_library, ONLY : set_print_library_iulog, print_vic_clock
+
+    IMPLICIT NONE
+
+    !--- INPUT/OUTPUT PARAMETERS:
+    TYPE(ESMF_Clock)            , INTENT(in)    :: EClock
+    TYPE(seq_cdata)             , INTENT(inout) :: cdata
+    TYPE(mct_aVect)             , INTENT(inout) :: x2l, l2x
+    TYPE(seq_cdata)             , INTENT(inout) :: cdata_s
+    TYPE(mct_aVect)             , INTENT(inout) :: x2s, s2x
+    CHARACTER(len=*), OPTIONAL  , INTENT(in)    :: NLFilename
+
+    !--- Local Variables
+    INTEGER                                     :: LNDID
+    INTEGER                                     :: mpicom_lnd
+    INTEGER                                     :: mytask, ierr
+    INTEGER                                     :: lsize, gsize
+    INTEGER                                     :: nx_global, ny_global
+    INTEGER                                     :: dtime  ! timestep (seconds)
+    INTEGER, POINTER                            :: idata(:)
+    INTEGER, DIMENSION(:), ALLOCATABLE          :: gindex  ! Number the local grid points
+    REAL(r8), POINTER                           :: lon_data(:)
+    REAL(r8), POINTER                           :: lat_data(:)
+    REAL(r8), POINTER                           :: area_data(:)
+    REAL(r8), POINTER                           :: mask_data(:)
+    REAL(r8), POINTER                           :: frac_data(:)
+    TYPE(mct_gsMap),  POINTER                   :: GSMap_lnd
+    TYPE(mct_gGrid),  POINTER                   :: dom_lnd
+    TYPE(seq_infodata_type), POINTER            :: infodata
+    CHARACTER(len=VICMAXSTRING)                 :: caseid
+    CHARACTER(len=VICMAXSTRING)                 :: starttype
+    CHARACTER(len=VICMAXSTRING)                 :: calendar
+    LOGICAL                                     :: exists  ! true if file exists
+    CHARACTER(len=*), PARAMETER                 :: subname = '(lnd_init_mct)'
+
+    !--- Local Variables for C/Fortran Interface
+    INTEGER(C_INT)                              :: errno
+    CHARACTER(len=VICMAXSTRING, kind=C_CHAR)    :: vic_global_param_file
+
+    !--- initialize the field index values, subroutine is below in this file
+    CALL cpl_indices_set()
+
+    !--- point/get data from cdata datatype
+    CALL seq_cdata_setptrs(cdata, ID=LNDID, mpicom=mpicom_lnd, &
+         gsMap=GSMap_lnd, dom=dom_lnd, infodata=infodata)
+    CALL MPI_COMM_RANK(mpicom_lnd, mytask, ierr)
```

Comment:
Interesting and yes, a bit confusing.  At least its not a bug.  I've moved on and everything seems to be okay at the MPI level. (We've got plenty of other problems!)